static/app/views/settings/projectSecurityHeaders/csp.tsx

```diff
@@ -8,76 +6,68 @@ import Panel from 'sentry/components/panels/panel';
-  organization: Organization;
-};
-
-type State = {
-  keyList: null | ProjectKey[];
-  project: null | Project;
-} & DeprecatedAsyncView['state'];
-
-export default class ProjectCspReports extends DeprecatedAsyncView<Props, State> {
-  getEndpoints(): ReturnType<DeprecatedAsyncView['getEndpoints']> {
-    const {organization} = this.props;
-    const {projectId} = this.props.params;
-    return [
-      ['keyList', `/projects/${organization.slug}/${projectId}/keys/`],
-      ['project', `/projects/${organization.slug}/${projectId}/`],
-    ];
-  }
-
-  getTitle() {
-    const {projectId} = this.props.params;
-    return routeTitleGen(t('Content Security Policy (CSP)'), projectId, false);
-  }
+function getInstructions(keyList: ProjectKey[]) {
+  return (
+    'def middleware(request, response):\n' +
+    "    response['Content-Security-Policy'] = \\\n" +
+    '        "default-src *; " \\\n' +
+    "        \"script-src 'self' 'unsafe-eval' 'unsafe-inline' cdn.example.com cdn.ravenjs.com; \" \\\n" +
+    "        \"style-src 'self' 'unsafe-inline' cdn.example.com; \" \\\n" +
+    '        "img-src * data:; " \\\n' +
+    '        "report-uri ' +
+    getSecurityDsn(keyList) +
+    '"\n' +
+    '    return response\n'
+  );
+}
 
-  getInstructions(keyList: ProjectKey[]) {
-    return (
-      'def middleware(request, response):\n' +
-      "    response['Content-Security-Policy'] = \\\n" +
-      '        "default-src *; " \\\n' +
-      "        \"script-src 'self' 'unsafe-eval' 'unsafe-inline' cdn.example.com cdn.ravenjs.com; \" \\\n" +
-      "        \"style-src 'self' 'unsafe-inline' cdn.example.com; \" \\\n" +
-      '        "img-src * data:; " \\\n' +
-      '        "report-uri ' +
-      getSecurityDsn(keyList) +
-      '"\n' +
-      '    return response\n'
-    );
-  }
+function getReportOnlyInstructions(keyList: ProjectKey[]) {
+  return (
+    'def middleware(request, response):\n' +
+    "    response['Content-Security-Policy-Report-Only'] = \\\n" +
+    '        "default-src \'self\'; " \\\n' +
+    '        "report-uri ' +
+    getSecurityDsn(keyList) +
+    '"\n' +
+    '    return response\n'
+  );
+}
 
-  getReportOnlyInstructions(keyList: ProjectKey[]) {
-    return (
-      'def middleware(request, response):\n' +
-      "    response['Content-Security-Policy-Report-Only'] = \\\n" +
-      '        "default-src \'self\'; " \\\n' +
-      '        "report-uri ' +
-      getSecurityDsn(keyList) +
-      '"\n' +
-      '    return response\n'
-    );
+export default function ProjectCspReports() {
+  const organization = useOrganization();
+  const params = useParams();
+  const projectId = params.projectId;
+
+  const {data: keyList} = useApiQuery<ProjectKey[]>(
+    [`/projects/${organization.slug}/${projectId}/keys/`],
+    {staleTime: 0}
+  );
+  const {data: project} = useApiQuery<Project>(
+    [`/projects/${organization.slug}/${projectId}/`],
+    {staleTime: 0}
+  );
+
+  if (!keyList || !project) {
+    return null;
```

Comment:
For this to be similar to async view that its replacing there should be a loading state and an error state

Reply:
yes, will add 👍 
I see the AsyncComponent has quite a bit of logic for the error handling.
Do we have a helper for that or is not necessary to do all that nowadays?

Reply:
i've been hooking up <LoadingError with the retry calling refetch and that's about it. There's still error handling at the router level

Reply:
https://www.notion.so/sentry/Guide-How-to-convert-AsyncComponent-and-AsyncView-to-useApiQuery-55751df48dc04df5b4b0da42ba5e4fbd?pvs=4#e771950dabea4588bb1f8ee555d67214